src/opera/pge/base_pge.py

```diff
@@ -295,6 +266,196 @@ def _finalize_log(self):
                          f"Closing log file {self.logger.get_file_name()}")
         self.logger.close_log_stream()
 
+    def _core_filename(self, inter_filename=None):
+        """
+        Returns the core file name component for products produced by the
+        Base PGE. This function should typically be overridden by inheritors
+        of PostProcessorMixin to accomplish the specific file-naming conventions
+        required by the PGE.
+
+        The core file name component of the Base PGE consists of:
+
+            <PROJECT>_<LEVEL>_<PGE NAME>_<TIMETAG>
+
+        Callers of this function are responsible for assignment of any other
+        product-specific fields, such as the file extension.
+
+        Parameters
+        ----------
+        inter_filename : str, optional
+            The intermediate filename of the output product to generate the
+            core filename for. This parameter may be used to inspect the file
+            in order to derive any necessary components of the returned filename.
+
+        Returns
+        -------
+        core_filename : str
+            The core file name component to assign to products created by this PGE.
+
+        """
+        time_tag = get_time_for_filename(self.production_datetime)
+
+        return f"{self.PROJECT}_{self.LEVEL}_{self.NAME}_{time_tag}"
+
+    def _geotiff_filename(self, inter_filename):
+        """
+        Returns the file name to use for GeoTIFF's produced by the Base PGE.
+
+        The GeoTIFF filename for the Base PGE consists of:
+
+            <Core filename>_<inter_filename>.tif
+
+        Where <Core filename> is returned by PostProcessorMixin._core_filename()
+
+        Parameters
+        ----------
+        inter_filename : str, optional
```

Comment:
Is this really option.  Won't line 323 fail if this is not defined?

Reply:
Good find, I definitely wanted the interface to allow `inter_filename` to be optional, but this function, and the version in `dsxw_hls.py` should also be checking for `None` and handling appropriately.